crates/edr_eth/src/remote/client.rs

```diff
@@ -1245,20 +1248,26 @@ mod tests {
 
         #[tokio::test]
         async fn call_bad_api_key() {
-            let alchemy_url = "https://eth-mainnet.g.alchemy.com/v2/abcdefg";
+            let api_key = "invalid-api-key";
+            let alchemy_url = format!("https://eth-mainnet.g.alchemy.com/v2/{api_key}");
 
             let hash = B256::from_str(
                 "0xc008e9f9bb92057dd0035496fbf4fb54f66b4b18b370928e46d6603933022222",
             )
             .expect("failed to parse hash from string");
 
-            let error = TestRpcClient::new(alchemy_url)
+            let error = TestRpcClient::new(&alchemy_url)
                 .call::<Option<eth::Transaction>>(RequestMethod::GetTransactionByHash(hash))
                 .await
                 .expect_err("should have failed to interpret response as a Transaction");
 
+            assert!(!error.to_string().contains(api_key));
+
             if let RpcClientError::HttpStatus(error) = error {
-                assert_eq!(error.status(), Some(StatusCode::from_u16(401).unwrap()));
+                assert_eq!(
+                    <ReqwestError as Into<reqwest::Error>>::into(error).status(),
```

Comment:
Same question w.r.t `From::from`

Reply:
fixed in [Simplify error conversion](https://github.com/NomicFoundation/hardhat/pull/5067/commits/fa0abb8e7574ecdeae2c4a4a6d6761d27ac52023)